astronomer/providers/apache/spark/triggers/spark_submit.py

```diff
@@ -0,0 +1,194 @@
+        self.polling_period_seconds = polling_period_seconds
+        self.application = application
+        self.conf = conf
+        self.conn_id = conn_id
+        self.files = files
+        self.py_files = py_files
+        self.archives = archives
+        self.driver_class_path = driver_class_path
+        self.jars = jars
+        self.java_class = java_class
+        self.packages = packages
+        self.exclude_packages = exclude_packages
+        self.repositories = repositories
+        self.total_executor_cores = total_executor_cores
+        self.executor_cores = executor_cores
+        self.executor_memory = executor_memory
+        self.driver_memory = driver_memory
+        self.keytab = keytab
+        self.principal = principal
+        self.proxy_user = proxy_user
+        self.name = name
+        self.num_executors = num_executors
+        self.application_args = application_args
+        self.env_vars = env_vars
+        self.verbose = verbose
+        self.spark_binary = spark_binary
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes SparkSubmitTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.apache.spark.triggers.spark_submit.SparkSubmitTrigger",
+            {
+                "task_id": self.task_id,
+                "polling_period_seconds": self.polling_period_seconds,
+                "application": self.application,
+                "conn_id": self.conn_id,
+                "name": self.name,
+                "conf": self.conf,
+                "files": self.files,
+                "py_files": self.py_files,
+                "archives": self.archives,
+                "driver_class_path": self.driver_class_path,
+                "jars": self.jars,
+                "java_class": self.java_class,
+                "packages": self.packages,
+                "exclude_packages": self.exclude_packages,
+                "repositories": self.repositories,
+                "total_executor_cores": self.total_executor_cores,
+                "executor_cores": self.executor_cores,
+                "executor_memory": self.executor_memory,
+                "driver_memory": self.driver_memory,
+                "keytab": self.keytab,
+                "principal": self.principal,
+                "proxy_user": self.proxy_user,
+                "num_executors": self.num_executors,
+                "application_args": self.application_args,
+                "env_vars": self.env_vars,
+                "verbose": self.verbose,
+                "spark_binary": self.spark_binary,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """
+        Makes a call to SparkSubmitHookAsync class and execute the spark-submit job in async mode
+        and based on the status it polls
+        """
+        hook = self._get_async_hook()
+        try:
+            await hook.spark_submit_job(self.application)
```

Comment:
I think we should not submit the job on the Triggerer. We should only track it. Job submission should happen on the Operator.